### ISSUE TITLE
User tokens with no set expiration date are automatically set to enabled: false [BUG]

### ISSUE THREAD

**Issue by emmeowzing** [rancher/rancher#42678]
**Rancher Server Setup**
- Rancher version: 2.7.5
- Installation option (Docker install/Helm Chart): Helm chart
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1
- Proxy/Cert Details: Certs were provisioned separately from a 3rd party provider.

**Information about the Cluster**
- Kubernetes version: 1.24.14
- Cluster Type (Local/Downstream): Local (GKE-based)
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider):

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: I'm an admin. This is irrelevant to this bug report.

**Describe the bug**

Users are creating tokens on a cluster and for some reason the `enabled`-field on these tokens is being automatically flipped from `true` to `false`, and I can't seem to diagnose why. Several users have reported this issue, I don't think it's due to any intentional action they are taking in the UI (or otherwise). They also do not have any access to the local cluster rancher is hosted on.

**To Reproduce**

I'm not sure how; I was hoping to glean some insight. Cf. my other issue, https://github.com/rancher/rancher/issues/42677, in case that's related.

**Result**

Users get 401's / Unauthorized errors via API when trying to reach our clusters through Rancher because their tokens have been disabled, not invalidated. If I edit their specific Token objects in K8s and flip this `enabled: false` back to `true` or delete the field entirely, they no longer get 401's.

**Expected Result**

Tokens shouldn't be automatically disabled without some clear indication of why.

**Screenshots**

N/A

**Additional context**

Semi-related, but not really: https://github.com/rancher/rancher/issues/15293
I've opened this bug at the same time as this one, not sure if they're perhaps related - https://github.com/rancher/rancher/issues/42677
https://github.com/rancher/rancher/blob/v2.7.6/pkg/apis/management.cattle.io/v3/authn_types.go#L34

**Comment by emmeowzing** [comment]
@rjeschmi we also use LDAP for authentication. I haven't noticed any sync issues, but I can try to keep a closer eye. Have you seen sync issues in a particular pod's logs?

**Comment by emmeowzing** [comment]
@rjeschmi what is the state of your pods?

https://github.com/rancher/rancher/issues/42677

we regularly have to fix these. I still wonder if it's related to perhaps the sync issues and correlated with this one.

**Comment by rjeschmi** [comment]
no, we don't have the eviction problem. I agree with it being a small number of users and one of those is me. I definitely have a large number of tokens and not all are problematic.

The sync complaints are group updates and likely related to upstream issues. I haven't correlated them exactly, but I'm suspicious.

**Comment by bgurung0** [comment]
Hi @emmeowzing, Were you able to resolve/bypass this issue? We are also using LDAP and noticing something similar.
